src/message/messageActionSheet.js

```diff
@@ -90,6 +91,46 @@ const muteTopic = async ({ auth, message }) => {
 muteTopic.title = 'Mute topic';
 muteTopic.errorMessage = 'Failed to mute topic';
 
+const deleteTopic = async ({ auth, message, dispatch, ownEmail, _ }) => {
+  const alertTitle = _.intl.formatMessage(
+    {
+      id: "Are you sure you want to delete the topic '{topic}'?",
+      defaultMessage: "Are you sure you want to delete the topic '{topic}'?",
+    },
+    { topic: message.subject },
+  );
+  const AsyncAlert = async () =>
+    new Promise((resolve, reject) => {
+      Alert.alert(
+        alertTitle,
+        _('This will also delete all messages in the topic.'),
+        [
+          {
+            text: _('Delete topic'),
+            onPress: async () => {
+              await dispatch(
+                deleteMessagesForTopic(getNarrowFromMessage(message, ownEmail), message),
+              ).catch(err => reject(err));
+              resolve('ok');
+            },
+            style: 'destructive',
+          },
+          {
+            text: _('Cancel'),
+            onPress: () => {
+              resolve('ok');
+            },
+            style: 'cancel',
+          },
+        ],
+        { cancelable: true },
+      );
+    });
```

Comment:
If instead you do nothing but `resolve(true)` and `resolve(false)` in the two `onPress` handlers, you can move all the application logic out of the interior by testing the return value of `await AsyncAlert`. This should result in nearly-symmetric branches in the `Alert.alert` call, more linear control flow for the whole function, and exception forwarding essentially for free.

(Also, `AsyncAlert` could use a type annotation.)

Reply:
Done!